### PR TITLE
Adding `intent` value to HoverHandler

### PR DESCRIPTION
**Pull request by jstnjns**
## Overview
To avoid wild actions when cursor moves across a grid where tiles are listing for hover, we've added `intent` so that you can listen to hover only when the user has stopped moving the mouse.

## Risks
None - This is adding functionality, not changing any.

### CODE REVIEW

**Comment by a-axton** [comment]
is it possible that the timeout could still exist if the component unmounts? should we clear the timer just incase?

**Comment by jstnjns** [comment]
> is it possible that the timeout could still exist if the component unmounts? should we clear the timer just incase? eg, if the component is also a link to another page/route

Good call.  Updating

File: src/components/HoverHandler/index.js [hunk 1]
```diff
@@ -2,34 +2,65 @@ import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
 
 
+const DELAY = 300
```
Comment (bernabe9) [inline src/components/HoverHandler/index.js]: What do you think of having a prop (with the default value 300) for the delay?

File: src/components/HoverHandler/index.js [hunk 2]
```diff
@@ -2,34 +2,65 @@ import React, { PureComponent } from 'react'
...
     children: PropTypes.func.isRequired,
   }
 
   state = {
     hovering: false,
+    intent: false,
+  }
+
+  componentWillUnmount () {
+    if (this.timer) this.timer = clearTimeout(this.timer)
```
Comment (bernabe9) [inline src/components/HoverHandler/index.js]: since the component is unmounting, do you need to assign the value of `clearTimeout` to `this.timer`?

```javascript
componentWillUnmount () {
  if (this.timer) clearTimeout(this.timer)
}
```

File: src/components/HoverHandler/index.js [hunk 3]
```diff
@@ -2,34 +2,65 @@ import React, { PureComponent } from 'react'
...
+    const {
+      hovering,
+      intent,
+    } = this.state
 
     return (
-      <div
-        onMouseEnter={this.onEnter.bind(this)}
-        onMouseLeave={this.onLeave.bind(this)}
+      <span
```
Comment (bernabe9) [inline src/components/HoverHandler/index.js]: just out for curiosity, why did you change it to a `span`?

File: src/components/HoverHandler/index.js (hunk 2 above)
Comment (jstnjns) [reply src/components/HoverHandler/index.js]: probably not

File: src/components/HoverHandler/index.js (hunk 1 above)
Comment (jstnjns) [reply src/components/HoverHandler/index.js]: thought about that, didn't really see a need

File: src/components/HoverHandler/index.js (hunk 3 above)
Comment (jstnjns) [reply src/components/HoverHandler/index.js]: so that this could wrap inline elements without having odd consequences or needing additional styling
